src/createActions.js

```diff
@@ -0,0 +1,59 @@
+import createAction from './createAction';
+
+function formatType(options, key) {
+  return `${options.prefix}${key}`;
+}
+
+/**
+   @param actionMap object optional - creates actions for each key,
+   using the value to determine the payload and meta creators. If
+   value is a fn, use it for the payloadCreator. If value is an
+   object use its `payload` and `meta` keys for payload and meta
+   creators.
+   @param arrayDefTypes array of strings optional
+   @param options object optional
+   @param options.prefix - string prefix all action types with, default ''
+   @returns object of actions
+*/
+export default function createActions(actionMap, arrDefTypes, options) {
+  // adjust for optional arguments
+  if (Array.isArray(actionMap)) { // actionMap not provided, shift
+    options = arrDefTypes;
+    arrDefTypes = actionMap;
+    actionMap = {};
+  }
+
+  if (!Array.isArray(arrDefTypes)) { // arrDefTypes not provided, shift
+    options = arrDefTypes;
+    arrDefTypes = [];
+  }
+
+  if (!options) { options = {}; }
+  if (!options.prefix) { options.prefix = ''; }
+
+  const mappedActions = Object.keys(actionMap).reduce(
+    (acc, key) => {
+      const type = formatType(options, key);
+      const action = actionMap[key];
+      let payloadCreator = undefined;
+      let metaCreator = undefined;
```

Comment:
I think something like this would be a bit simpler/clearer:

```js
if (isFunction(action)) {
  acc[key] = createAction(type, action);
} else if (isPlainObject(action)) {
  acc[key] = createAction(type, action.payload, action.meta);
}
```

`acc` could also probably be named better. We all know what it abbreviates ("accumulator"), but maybe `mappedActionsSoFar` makes more sense in this specific context?